drivers/timer/Kconfig.cortex_m_systick

```diff
@@ -24,3 +24,15 @@ config CORTEX_M_SYSTICK_INSTALL_ISR
 	help
 	  This option should be selected by SysTick-based drivers so that the
 	  sys_clock_isr() function is installed.
+
+if CORTEX_M_SYSTICK
+config CORTEX_M_SYSTICK_EXT_CLOCK
+	bool "Use external reference clock as SysTick source instead of CPU one"
+	default n
+	depends on CORTEX_M_SYSTICK
+
+config CORTEX_M_SYSTICK_EXT_FRQ
+	int "External clock frequency"
+	depends on CORTEX_M_SYSTICK_EXT_CLOCK
```

Comment:
Traditionally timer drivers have been only loosely coupled to devicetree (it's just history, they're very low level and they predate DTS), but as we add new configurables it's probably best to do it where it belongs.